File: 0896-monotonic-array/0896-monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        if(nums.size() == 1) return true;
        int n = 0;
        for(int i = 0; i < nums.size() - 1; i++){
            if(nums[i] <= nums[i+1]) n++;
        }
        if(n == nums.size() - 1) return true;
        n = 0;
        for(int i = 0; i < nums.size() - 1; i++){
            if(nums[i] >= nums[i+1]) n++;
        }
        if(n == nums.size() - 1) return true;
        return false;
    }

    //  ==============================  //
    //  제일 빠른 코드
    // bool isMonotonic(vector<int>& nums) {
    //     cin.tie(0);
    //     ios::sync_with_stdio(0);
    //     return isminc(nums)||ismaxc(nums);
    // }
    // bool isminc(vector<int>& nums){
    //     for (int i=0;i<nums.size()-1;i++){
    //         if (nums[i]>nums[i+1]){
    //             return false;
    //         }
    //     }
    //     return true;
    // }
    // bool ismaxc(vector<int>& nums){
    //     for (int i=0;i<nums.size()-1;i++){
    //         if (nums[i]<nums[i+1]){
    //             return false;
    //         }
    //     }
    //     return true;
    // }
};

File: 0896-monotonic-array/0896-monotonic-array.cs
public class Solution {
    public bool IsMonotonic(int[] nums) {
        //  https://leetcode.com/problems/monotonic-array/solutions/4102869/95-15-single-pass-check
        if(nums.Length < 2) return true;

        int direction = 0;

        for(int i = 1; i < nums.Length; i++){
            if(nums[i] > nums[i-1]){
                if(direction == 0) direction = 1;
                else if(direction == -1) return false;
            }else if(nums[i] < nums[i-1]){
                if(direction == 0) direction = -1;
                else if(direction == 1) return false;
            }
        }

        return true;
    }
}